### DISCUSSION TITLE
Add control button to mjpeg_server.py or mjpeg_server_2.py

### DISCUSSION THREAD

**Discussion by ballen4705** [raspberrypi/picamera2#613] [Q&A] [answered]
I am using this code:
https://github.com/raspberrypi/picamera2/blob/main/examples/mjpeg_server_2.py
to stream picamera2 output to a web browser.

I'd like to add a button on the web page, which allows the user sitting at the web browser to toggle an on/off variable or call a function in the server.  When they push this button, it should not interrupt the streaming video window on the browser.

Could someone share a bit of code that does something similar?  (A few lines of embedded java might be needed to avoid reloading the page and/or interrupting the video, but I am not sure.)

Thanks,
  Bruce

**Comment by sandyol55** [comment, marked as answer]
The code in
https://www.e-tinkers.com/2018/04/how-to-control-raspberry-pi-gpio-via-http-web-server/
may be a useful starting point.

The use of the out of date path /opt/vc/bin/ can be removed to call vcgencmd directly to keep the example code running correctly in current releases.

**Reply by ballen4705** [reply]
Thank you, that is exactly what I was looking for! The final javascript solution looks clean and simple. I’ll try this.
